packages/expo-image-manipulator/ios/ImageManipulatorModule.swift

```diff
@@ -0,0 +1,133 @@
+
+  public func definition() -> ModuleDefinition {
+    name("ExpoImageManipulator")
+
+    function("manipulateAsync", manipulateImage)
+      .runOnQueue(.main)
+  }
+
+  internal func manipulateImage(url: URL, actions: [ManipulateAction], options: ManipulateOptions, promise: Promise) {
+    loadImage(atUrl: url) { result in
+      switch result {
+      case .failure(let error):
+        return promise.reject(error)
+      case .success(let image):
+        do {
+          let newImage = try manipulate(image: image, actions: actions)
+          let saveResult = try self.saveImage(newImage, options: options)
+
+          promise.resolve([
+            "uri": saveResult.url.absoluteString,
+            "width": newImage.cgImage?.width ?? 0,
+            "height": newImage.cgImage?.height ?? 0,
+            "base64": options.base64 ? saveResult.data.base64EncodedData() : nil
+          ])
+        } catch {
+          promise.reject(error)
+        }
+      }
+    }
+  }
+
+  /**
+   Loads the image from given URL.
+   */
+  internal func loadImage(atUrl url: URL, callback: @escaping LoadImageCallback) {
+    if url.scheme == "data" {
+      guard let data = try? Data(contentsOf: url), let image = UIImage(data: data) else {
+        return callback(.failure(CorruptedImageDataError()))
+      }
+      return callback(.success(image))
+    }
+    if url.scheme == "assets-library" {
+      // TODO: ALAsset URLs are deprecated as of iOS 11, we should migrate to `ph://` soon.
+      return loadImageFromPhotoLibrary(url: url, callback: callback)
+    }
+
+    guard let imageLoader = self.appContext?.imageLoader else {
+      return callback(.failure(ImageLoaderNotFoundError()))
+    }
+    guard let fileSystem = self.appContext?.fileSystem else {
+      return callback(.failure(FileSystemNotFoundError()))
+    }
+    guard fileSystem.permissions(forURI: url).contains(.read) else {
+      return callback(.failure(FileSystemReadPermissionError(path: url.absoluteString)))
+    }
+
+    imageLoader.loadImage(for: url) { error, image in
+      guard let image = image, error == nil else {
+        return callback(.failure(ImageLoadingFailedError(cause: error.debugDescription)))
+      }
+      callback(.success(image))
+    }
+  }
+
+  /**
+   Loads the image from user's photo library.
+   */
+  internal func loadImageFromPhotoLibrary(url: URL, callback: @escaping LoadImageCallback) {
+    guard let asset = PHAsset.fetchAssets(withALAssetURLs: [url], options: nil).firstObject else {
+      return callback(.failure(ImageNotFoundError()))
+    }
+    let size = CGSize(width: asset.pixelWidth, height: asset.pixelHeight)
+    let options = PHImageRequestOptions()
+
+    options.resizeMode = .exact
+    options.isNetworkAccessAllowed = true
+    options.isSynchronous = true
+    options.deliveryMode = .highQualityFormat
+
+    PHImageManager.default().requestImage(for: asset, targetSize: size, contentMode: .aspectFit, options: options) { image, info in
```

Comment:
**⚠️  SwiftLint Violation: Unused Closure Parameter**
Unused parameter "info" in a closure should be replaced with _.